mne/viz/_3d.py

```diff
@@ -1143,3 +1143,132 @@ def plot_dipole_locations(dipoles, trans, subject, subjects_dir=None,
         fig.scene.x_plus_view()
 
     return fig
+
+
+def snapshot_brain_montage(fig, montage, hide_sensors=True):
+    """Take a snapshot of a Mayavi Scene and project channels onto 2d coords.
+
+    Note that this will take the raw values for 3d coordinates of each channel,
+    without applying any transforms. If brain images are flipped up/dn upon
+    using `imshow`, check your matplotlib backend as this behavior changes.
+
+    Parameters
+    ----------
+    fig : instance of Mayavi Scene
+        The figure on which you've plotted electrodes using `plot_trans`.
+    montage : instance of `DigMontage` or `Info` | dict of ch: xyz mappings.
+        The digital montage for the electrodes plotted in the scene. If `Info`,
+        channel positions will be pulled from the `loc` field of `chs`.
+    hide_sensors : bool
+        Whether to remove the spheres in the scene before taking a snapshot.
+
+
+    Returns
+    -------
+    xy : array, shape (n_channels, 2)
+        The 2d location of each channel on the image of the current scene view.
+    im : array, shape (m, n, 3)
+        The screenshot of the current scene view
+    """
+    from mayavi import mlab
+    from ..channels import Montage, DigMontage
+    from .. import Info
+    if isinstance(montage, (Montage, DigMontage)):
+        xyz = np.vstack(montage.dig_ch_pos.values())
+        ch_names = montage.dig_ch_pos.keys()
+    elif isinstance(montage, Info):
+        xyz = np.vstack([ich['loc'][:3] for ich in montage['chs']])
+        ch_names = np.array([ich['ch_name'] for ich in montage['chs']])
+    elif isinstance(montage, dict):
+        if not all(len(ii) == 3 for ii in montage.values()):
+            raise ValueError('All electrode positions must be length 3')
+        ch_names, xyz = zip(*[(ich, ixyz) for ich, ixyz in montage.items()])
+        xyz = np.vstack(xyz)
+    else:
+        raise ValueError('montage must be an instance of `DigMontage`, `Info`,'
+                         ' or `dict`')
+
+    xy = _3d_to_2d(fig, xyz)
+    xy = dict(zip(ch_names, xy))
+    pts = fig.children[-1]
+
+    if hide_sensors is True:
+        pts.visible = False
+    im = mlab.screenshot(fig)
+    pts.visible = True
+    return xy, im
```

Comment:
you'll need to pass back `ch_names` because the order is non-deterministic, right?